packages/core/src/shipping/strategies/braintree/braintree-accelerated-checkout-shipping-strategy.ts

```diff
@@ -0,0 +1,219 @@
+} from '../../../common/error/errors';
+import { CustomerAddress } from '../../../customer';
+import { Country } from '../../../geography';
+import { PaymentMethodActionCreator } from '../../../payment';
+import { PaymentProviderCustomerActionCreator } from '../../../payment-provider-customer';
+import { CardInstrument } from '../../../payment/instrument';
+import { UntrustedShippingCardVerificationType } from '../../../payment/instrument/instrument';
+import ConsignmentActionCreator from '../../consignment-action-creator';
+import { ShippingRequestOptions } from '../../shipping-request-options';
+import ShippingStrategy from '../shipping-strategy';
+
+import BraintreeAcceleratedCheckoutInitializeOptions from './braintree-accelerated-checkout-initialize-options';
+
+export default class BraintreeAcceleratedCheckoutShippingStrategy implements ShippingStrategy {
+    private _browserStorage: BrowserStorage;
+
+    constructor(
+        private store: CheckoutStore,
+        private billingAddressActionCreator: BillingAddressActionCreator,
+        private consignmentActionCreator: ConsignmentActionCreator,
+        private paymentMethodActionCreator: PaymentMethodActionCreator,
+        private paymentProviderCustomerActionCreator: PaymentProviderCustomerActionCreator,
+        private braintreeIntegrationService: BraintreeIntegrationService,
+    ) {
+        this._browserStorage = new BrowserStorage('paypalConnect');
+    }
+
+    updateAddress(
+        address: AddressRequestBody,
+        options?: ShippingRequestOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        return this.store.dispatch(this.consignmentActionCreator.updateAddress(address, options));
+    }
+
+    selectOption(
+        optionId: string,
+        options?: ShippingRequestOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        return this.store.dispatch(
+            this.consignmentActionCreator.selectShippingOption(optionId, options),
+        );
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this.store.getState());
+    }
+
+    async initialize(
+        options: BraintreeAcceleratedCheckoutInitializeOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" argument is not provided.',
+            );
+        }
+
+        if (!this.shouldRunAuthenticationFlow()) {
+            return Promise.resolve(this.store.getState());
+        }
+
+        try {
+            await this.store.dispatch(this.paymentMethodActionCreator.loadPaymentMethod(methodId));
+            await this.runAuthenticationFlowOrThrow(methodId);
+        } catch (error) {
+            // Info: we should not throw any error here to avoid
+            // customer stuck on shipping step due to the payment provider
+            // custom flow
+        }
```

Comment:
What if an error occurs during `runAuthenticationFlowOrThrow`, can a shopper proceed to the payment and check out?

Reply:
I mean will the payment step work if the shipping encounters an error?

Reply:
Yes, the regular flow will work in this case.
The main idea of this empty `catch` is to skip all fails of AXO flow and not to lock checkout flow for shoppers.
And payment step also has logic for such case.